### PR TITLE
Refresh machete file annotations when repo changed

### PR DESCRIPTION
**Pull request by micpiotrowski**

### CODE REVIEW

**Review by mkondratek — APPROVED** [review]

**Comment by PawelLipski** [comment]
Whooops... just got the following error, most likely right after sliding in a new branch while editor was opened:

```
com.intellij.diagnostic.PluginException: Element: class com.virtuslab.gitmachete.frontend.file.grammar.MacheteFile #Machete  because: different providers: com.virtuslab.gitmachete.frontend.file.MacheteFileViewProvider{vFile=file:///home/plipski/git-machete-intellij-plugin/.git/machete, vFileId=399317, content=VirtualFileContent{size=17}, eventSystemEnabled=true}(13a00cc6); com.virtuslab.gitmachete.frontend.file.MacheteFileViewProvider{vFile=file:///home/plipski/git-machete-intellij-plugin/.git/machete, vFileId=399317, content=VirtualFileContent{size=17}, eventSystemEnabled=true}(7ab5c6c4)
invalidated at: see attachment [Plugin: com.virtuslab.git-machete]
	at com.intellij.ide.plugins.PluginManagerCore.createPluginException(PluginManagerCore.java:335)
	at com.intellij.diagnostic.PluginProblemReporterImpl.createPluginExceptionByClass(PluginProblemReporterImpl.java:12)
	at com.intellij.diagnostic.PluginException.createByClass(PluginException.java:68)
	at com.intellij.psi.util.PsiUtilCore.ensureValid(PsiUtilCore.java:478)
	at com.intellij.psi.impl.source.PsiFileImpl.getTextLength(PsiFileImpl.java:324)
	at com.intellij.codeInsight.daemon.impl.DaemonCodeAnalyzerImpl.restart(DaemonCodeAnalyzerImpl.java:557)
	at com.virtuslab.gitmachete.frontend.file.MacheteFileViewProvider.lambda$subscribeToGitRepositoryChanges$0(MacheteFileViewProviderFactory.java:47)
	at com.intellij.openapi.application.TransactionGuardImpl.runWithWritingAllowed(TransactionGuardImpl.java:216)
	at com.intellij.openapi.application.TransactionGuardImpl.access$200(TransactionGuardImpl.java:24)
	at com.intellij.openapi.application.TransactionGuardImpl$2.run(TransactionGuardImpl.java:199)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:822)
	at com.intellij.openapi.application.impl.ApplicationImpl.lambda$invokeLater$4(ApplicationImpl.java:324)
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:85)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:134)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:47)
	at com.intellij.openapi.application.impl.FlushQueue$FlushNow.run(FlushQueue.java:190)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:776)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:746)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:976)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:843)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:454)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:773)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$9(IdeEventQueue.java:453)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:822)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:501)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
Caused by: com.intellij.psi.PsiInvalidElementAccessException: Element: class com.virtuslab.gitmachete.frontend.file.grammar.MacheteFile #Machete  because: different providers: com.virtuslab.gitmachete.frontend.file.MacheteFileViewProvider{vFile=file:///home/plipski/git-machete-intellij-plugin/.git/machete, vFileId=399317, content=VirtualFileContent{size=17}, eventSystemEnabled=true}(13a00cc6); com.virtuslab.gitmachete.frontend.file.MacheteFileViewProvider{vFile=file:///home/plipski/git-machete-intellij-plugin/.git/machete, vFileId=399317, content=VirtualFileContent{size=17}, eventSystemEnabled=true}(7ab5c6c4)
invalidated at: see attachment
	... 33 more
```

**Review by mkondratek — CHANGES_REQUESTED** [review]
🙅

**Comment by PawelLipski** [comment]
...yeah, the above seems to show up pretty consistently, now I've spotted it after Slide Out as well :/

File: frontendFile/src/main/java/com/virtuslab/gitmachete/frontend/file/MacheteFileViewProviderFactory.java [hunk 1]
```diff
@@ -17,17 +26,29 @@ public FileViewProvider createFileViewProvider(VirtualFile file,
...
   protected boolean shouldCreatePsi() {
     return true;
   }
+
+  private void subscribeToGitRepositoryChanges(Project project, Language language) {
+    Topic<GitRepositoryChangeListener> topic = GitRepository.GIT_REPO_CHANGE;
+    GitRepositoryChangeListener listener = repository -> GuiUtils.invokeLaterIfNeeded(() -> {
+      PsiFile psiFile = getPsi(language);
+      if (psiFile != null) {
+        FileContentUtil.reparseFiles(project, java.util.List.of(psiFile.getVirtualFile()), true);
```
Comment (PawelLipski) [inline frontendFile/src/main/java/com/virtuslab/gitmachete/frontend/file/MacheteFileViewProviderFactory.java:49]: Pls name the `true` param, it's unclear what it means :thinking:

**Review by PawelLipski — APPROVED** [review]
Tested locally, works as expected :ok_hand:

**Review by PawelLipski — APPROVED** [review]

**Review by mkondratek — APPROVED** [review]

**Pull request merged**